src/main/scala/com/stratio/khermes/cluster/collector/CommandCollectorActor.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2016 Stratio (http://stratio.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.stratio.khermes.cluster.collector
+
+import java.util.UUID
+
+import akka.actor.{ActorLogging, Props}
+import akka.cluster.MemberStatus
+import akka.cluster.pubsub.DistributedPubSub
+import akka.cluster.pubsub.DistributedPubSubMediator.Publish
+import akka.stream.actor.ActorPublisher
+import com.stratio.khermes.cluster.collector.CommandCollectorActor.Check
+import com.stratio.khermes.cluster.supervisor.NodeSupervisorActor
+import com.stratio.khermes.cluster.supervisor.NodeSupervisorActor.Result
+
+import scala.concurrent.duration._
+import scala.util.Try
+
+class CommandCollectorActor extends ActorPublisher[CommandCollectorActor.Result] with ActorLogging {
+
+  val MaxCommandTimeout = 10L
+  val CheckCommandStateTimeout = 100 milliseconds
+  val mediator = DistributedPubSub(context.system).mediator
+
+  var commands = scala.collection.mutable.HashMap.empty[String, (Long, List[Result])]
```

Comment:
It would be nice to avoid vars. please review this [post](https://scalerablog.wordpress.com/2017/03/20/scalera-tip-keep-your-actors-state-with-no-var-at-all/) and tell what do you think.